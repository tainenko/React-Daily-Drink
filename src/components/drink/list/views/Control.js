import React from "react";

const Control = ({isDelClickable, handleAddItem, handleDeleteBtn}) => (
    <div className="dailyDrink__list__control">
        <button
            data-testid="add button"
            className="btn btn-add transition"
            onClick={handleAddItem}
        >
            + Add
        </button>
        <button
            data-testid="del button"
            className={`btn btn-delete transition ${isDelClickable ? "" : "disable"}`}
            onClick={handleDeleteBtn}
        >
            - Delete
        </button>
    </div>
);

export default Control;