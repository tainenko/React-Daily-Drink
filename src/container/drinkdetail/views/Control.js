import React from "react";
import {useFormikContext} from "formik";

const Control = () => {
    const {values, isSubmitting} = useFormikContext();
    return (<div className="dailyDrink__orderList__control">
            <button type="submit"
                    className={`btn btn-add transition ${!!(values.name && values.price) ? "" : "disable"}`}
                    disabled={isSubmitting}>
                Submit
            </button>
            <button type="button" className="btn btn-delete transition" onClick={values.handleCancel}>Cancel</button>
        </div>
    )
};


export default Control;