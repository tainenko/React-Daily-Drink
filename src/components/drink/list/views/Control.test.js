import React from "react";
import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Control from "./Control";

afterEach(cleanup);

const setup = () => {
    const utils = render(<Control isDelClickable={false} />);
    const addBtn = utils.getByTestId("add button");
    const delBtn = utils.getByTestId("del button");

    return {
        addBtn,
        delBtn,
        ...utils
    }
};

it('should take a snapshot', () => {
    const {asFragment} = setup();
    expect(asFragment()).toMatchSnapshot()
});

it('addBtn should have text "+ Add"',()=>{
    const {addBtn}=setup();
    expect(addBtn).toHaveTextContent("+ Add")
});

it('addBtn should have be enable',()=>{
    const {addBtn}=setup();
    expect(addBtn).toBeEnabled();
});

it('delBtn should have text "- Del"',()=>{
    const {delBtn}=setup();
    expect(delBtn).toHaveTextContent("- Del")
});

it('delBtn should be disable',()=>{
    const {delBtn}=setup();
    expect(delBtn).toBeDisabled()
});

it('delBtn should be enable when isDelClickable is true',()=>{
    const delBtn =render(<Control isDelClickable={true} />).getByTestId("del button");
    expect(delBtn).toBeEnabled();
});
