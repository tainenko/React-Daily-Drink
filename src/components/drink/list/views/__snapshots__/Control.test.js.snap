// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="dailyDrink__list__control"
  >
    <button
      class="btn btn-add transition"
      data-testid="add button"
    >
      + Add
    </button>
    <button
      class="btn btn-delete transition disable"
      data-testid="del button"
      disabled=""
    >
      - Delete
    </button>
  </div>
</DocumentFragment>
`;
